nucypher/utilities/networking.py

```diff
@@ -114,14 +114,15 @@ def get_external_ip_from_default_teacher(network: str,
                                          log: Logger = IP_DETECTION_LOGGER
                                          ) -> Union[str, None]:
 
-    # Prevents circular import
+    # Prevents circular imports
     from nucypher.characters.lawful import Ursula
+    from nucypher.network.nodes import TEACHER_NODES
 
     if federated_only and registry:
         raise ValueError('Federated mode must not be true if registry is provided.')
     base_error = 'Cannot determine IP using default teacher'
     try:
-        top_teacher_url = RestMiddleware.TEACHER_NODES[network][0]
+        top_teacher_url = TEACHER_NODES[network][0]
```

Comment:
In this function should we also be trying the next entry if the first one fails i.e.  if `NodeSeemToBeDown` L144. i.e. use a for-loop for all teachers available in TEACHER_NODES[network] and as soon as one is able to be connected to, L141, we short circuit the for-loop. I know it doesn't do that now, but we didn't have more than one entry before.

Or am I totally missing the mark?

^ @KPrasch / @cygnusv ?

Reply:
You are totally correct!  I missed the `[0]`